algosdk/template.py

```diff
@@ -175,6 +175,115 @@ def get_program(self):
         return inject(orig, offsets, values, types)
 
 
+class DynamicFee(Template):
+    """
+    DynamicFee contract allows you to create a transaction without
+    specifying the fee. The fee will be determined at the moment of
+    transfer.
+
+    Args:
+        receiver (str): address to receive the assets
+        amount (int): amount of assets to transfer
+        first_valid (int): first valid round for the transaction
+        last_valid (int, optional): last valid round for the transaction
+            (default to first_valid + 1000)
+        close_remainder_address (str, optional): if you would like to
+            close the account after the transfer, specify the address that
+            would recieve the remainder
+    """
+
+    def __init__(self, receiver: str, amount: int, first_valid: int,
+                 last_valid: int = None, close_remainder_address: str = None):
+        self.lease_value = bytes([random.randint(0, 255) for x in range(
+                                 constants.lease_length)])
+
+        self.last_valid = last_valid
+        if last_valid is None:
+            last_valid = first_valid + 1000
+
+        self.amount = amount
+        self.first_valid = first_valid
+        self.close_remainder_address = close_remainder_address
+        if close_remainder_address is None:
+            self.close_remainder_address = bytes(constants.address_len)
+        self.receiver = receiver
+        
+    def get_program(self):
+        """
+        Return a byte array to be used in LogicSig.
+        """
+        orig = ("ASAFAgEFBgcmAyD+vKC7FEpaTqe0OKRoGsgObKEFvLYH/FZTJclWlfaiEy" +
+                "DmmpYeby1feshmB5JlUr6YI17TM2PKiJGLuck4qRW2+QEGMgQiEjMAECMS" +
+                "EDMABzEAEhAzAAgxARIQMRYjEhAxECMSEDEHKBIQMQkpEhAxCCQSEDECJR" +
+                "IQMQQhBBIQMQYqEhA=")
+        orig = base64.b64decode(orig)
+        offsets = [5, 6, 7, 11, 44, 76]
+        values = [self.amount, self.first_valid, self.last_valid,
+                  self.receiver, self.close_remainder_address,
+                  base64.b64encode(self.lease_value)]
+        types = [int, int, int, "address", "address", "base64"]
+        return inject(orig, offsets, values, types)
+
+    @staticmethod
+    def get_transactions(txn, lsig, private_key, fee, first_valid, last_valid):
```

Comment:
Consider making the last valid an optional field and if `None` just do `first_valid+1000`